lib/chef/azure/commands/enable.rb

```diff
@@ -92,7 +114,24 @@ def install_chef_service
   end
 
   def enable_chef_service
-    @exit_code, error_message = ChefService.new.enable(@chef_extension_root, bootstrap_directory, @azure_plugin_log_location)
+    chef_service_interval = load_chef_service_interval
+
+    if chef_service_interval.empty?
+      @exit_code, error_message = ChefService.new.enable(@chef_extension_root, bootstrap_directory, @azure_plugin_log_location)
+    else
+      if chef_service_interval.to_i == 0
+        if ChefService.new.is_running?
+          puts "#{Time.now} Deleting the chef-client service on user's choice..."
+          ChefService.new.delete_cron
+        else
+          puts "#{Time.now} Not deploying the chef-client service on user's choice..."
+        end
+        @exit_code = 0
+      else
+        @exit_code, error_message = ChefService.new.enable(@chef_extension_root, bootstrap_directory, @azure_plugin_log_location, chef_service_interval.to_i)
+      end
+    end
+

```

Comment:
looking at the `ChefServer` again, it would probably be better to make these methods class level.